incl char to `set[char]` generates invalid C code

I got compile error from GCC when compiled following code:
```nim
proc foo: char = 'a'
var x: set[char] = {}
x.incl foo()
```
Error message:
```
nimcache/tesm7ycc51.c: In function 'main':
nimcache/tesm7ycc51.c:436:47: error: invalid initializer
  436 | AarrayAuS8ZS32_0_t X60Qdesugar_1_tesm7ycc51 = foo_0_tesm7ycc51();
      |                                               ^~~~~~~~~~~~~~~~
nimcache/tesm7ycc51.c:437:1: error: aggregate value used where an integer was expected
  437 | (*X60Qdesugar_0_tesm7ycc51).a[((NU)(((NU)X60Qdesugar_1_tesm7ycc51) >> 3ull))] = ((NU8)((*X60Qdesugar_0_tesm7ycc51).a[((NU)(((NU)X60Qd
esugar_1_tesm7ycc51) >> 3ull))] | ((NU8)(1ull << ((NU)(X60Qdesugar_1_tesm7ycc51 & 7ull))))));
      | ^
nimcache/tesm7ycc51.c:437:1: error: aggregate value used where an integer was expected
nimcache/tesm7ycc51.c:437:214: error: invalid operands to binary & (have 'AarrayAuS8ZS32_0_t' and 'long long unsigned int')
  437 | Qdesugar_0_tesm7ycc51).a[((NU)(((NU)X60Qdesugar_1_tesm7ycc51) >> 3ull))] | ((NU8)(1ull << ((NU)(X60Qdesugar_1_tesm7ycc51 & 7ull))))));
      |                                                                                                                          ^
```

incl character literal doesn't generates the error:
```nim
var x: set[char] = {}
x.incl 'a'
```

I found this issue when I tried to port `multiReplace` func in Nim 2 to Nimony.

--- a/src/hexer/desugar.nim
+++ b/src/hexer/desugar.nim
@@ -628,7 +628,7 @@ proc genInclExcl(c: var Context; dest: var TokenBuf; n: var Cursor) =
     dest.add parLeToken(StmtsS, info)
     # lift both so (n, (n = 123; n)) works
     a = liftTempAddr(c, dest, aOrig, typ, info)
-    b = liftTemp(c, dest, bOrig, typ, info)
+    b = liftTemp(c, dest, bOrig, typ.firstSon, info)
   else:
     a = aOrig
     b = bOrig

--- a/tests/nimony/sets/tsets.nif
+++ b/tests/nimony/sets/tsets.nif
@@ -282,12 +282,50 @@
      (discard 8
       (setconstr 2,~12
        (set 1
-        (c +8)) 1 'a')))))) 5,2
+        (c +8)) 1 'a')))))) ,42
+ (block . 2,1
+  (stmts
+   (proc 5 :foo.0 . . . 8
+    (params) 2
+    (c +8) . . 17
+    (stmts
+     (result :result.0 . . ~15
+      (c +8) .)
+     (asgn result.0 'a') ~17
+     (ret result.0))) 4,1
+   (var :x.0 . . 8,~17
+    (set 1
+     (c +8)) 15
+    (setconstr ~7,~17
+     (set 1
+      (c +8)))) 1,2
+   (incl
+    (set
+     (c +8)) ~1 x.0 9
+    (call ~3 foo.0)) 2,4,lib/std/assertions.nim
+   (stmts
+    (if 3
+     (elif
+      (not 11,48,tests/nimony/sets/tsets.nim
+       (eqset
+        (set
+         (c +8)) ~2 x.0 3
+        (setconstr ,~20
+         (set 1
+          (c +8)) 1 'a'))) ~1,1
+      (stmts 2,116,lib/std/syncio.nim
+       (stmts 2,1
+        (stmts
+         (cmd write.0.syn1lfpjv 6 stdout.0.syn1lfpjv 9,5,lib/std/assertions.nim "\5BAssertion Failure\5D ")) 2,1
+        (stmts
+         (cmd write.0.syn1lfpjv 6 stdout.0.syn1lfpjv 35,3,lib/std/assertions.nim "")) ,2
+        (cmd write.5.syn1lfpjv 6 stdout.0.syn1lfpjv 14 '\0A')) ,1
+       (cmd quit.0.syn1lfpjv 5 +1))))))) 5,2
  (proc :dollar`.Foo.0.tseflljd71 x . .
   (params
    (param :e.0 . . Foo.0.tseflljd71 .)) string.0.sysvq0asl . .
   (stmts 6
-   (result :result.0 . . string.0.sysvq0asl .) 6
+   (result :result.1 . . string.0.sysvq0asl .) 6
    (case e.0 ~9,1
     (of
      (ranges A.0.tseflljd71)
@@ -313,4 +351,4 @@
      (ranges F.0.tseflljd71)
      (stmts
       (ret "F"))))
-   (ret result.0))))
+   (ret result.1))))

--- a/tests/nimony/sets/tsets.nim
+++ b/tests/nimony/sets/tsets.nim
@@ -38,3 +38,11 @@ block:
 block:
   proc test() =
     discard {'a'}
+
+# issue #1459
+block:
+  proc foo: char = 'a'
+  var x: set[char] = {}
+  x.incl foo()
+
+  assert x == {'a'}

